### DISCUSSION TITLE
unable to run migrations in an empty PostGIS db

### DISCUSSION THREAD

**Discussion by benknab** [prisma/prisma#4732] [Q&A] [answered]
I'm struggling to use the migration tool after I upgraded to 2.13.0.
I have a few spatial columns that I manage with scripts besides the Prisma migration tool.

My pipeline used to be this:
1. apply Prisma migrations
2. run manual migrations and add / alter spatial columns if needed

but with 2.13.0, I can no longer run any Prisma migration on a fresh PostGIS database because of this error:
```
Error: P3005

The database schema for `localhost:5432` is not empty. Read more about how to baseline an existing production database: https://pris.ly/d/migrate-baseline
```

I'm also a Nexus and nexus-plugin-prisma user which doesn't support 2.12.x, so I can't downgrade reliably.

Is there a workaround for this?
I appreciate the help!

**Comment by tomhoule** [comment]
Hi @bkiac! Normally, this message would point to a use-case for baselining: https://www.prisma.io/docs/guides/prisma-guides/prisma-migrate-guides/add-prisma-migrate-to-a-project#baseline-your-production-environment

However if the database is empty, this is a bug. Can you please open an issue with as much detail as possible, so we can look into it?

**Reply by NMinhNguyen** [reply]
@tomhoule reading https://www.prisma.io/docs/guides/prisma-guides/prisma-migrate-guides/add-prisma-migrate-to-a-project#reset-other-development-environments

> 1. Check out a copy of the repository with the new migration directory and the initial migration
> 2. Run the following command to reset the development database:
>
>    ```terminal
>    prisma migrate dev  --preview-feature
>    ```

it gave me the impression that if I have already committed a new migration directory, it should give me the opportunity to reset an existing dev database regardless of whether it's empty. This is because when I ran the above command after deleting `prisma/migrations`, it prompted me to reset an existing non-empty dev database. However, now I'm getting the exact same `P3005` error. So either the docs are a bit misleading, or there is a bug. This discussion was the only related thing that I found so I decided to comment here instead of opening a new issue.

**Reply by tomhoule** [reply]
I think your issue is something we noticed as well, it is going to be fixed.

**Comment by benknab** [comment, marked as answer]
@tomhoule
I think it's caused by the PostGIS extension and since Prisma does not support it yet, I didn't open a bug.
Adding the PostGIS extension creates a `spatial_ref_sys` table, therefore schema is not empty, so Prisma is right.
If I reset the schema, migrations run successfully, but I lose the spatial types.
I created the issue here: https://github.com/prisma/prisma/issues/4745

**Reply by tomhoule** [reply]
Thanks for creating the issue — it will definitely be taken into account.

**Comment by quantuminformation** [comment]
still no PostGis support?

**Reply by ludralph** [reply]
Hi @quantuminformation, there is an open issue for supporting PostGIS. 

- https://github.com/prisma/prisma/issues/2789

Also, there are workarounds detailed in this [article](https://freddydumont.com/blog/prisma-postgis) and the issue itself.

**Reply by quantuminformation** [reply]
thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
